lib/pbench/server/cache_manager.py

```diff
@@ -226,6 +238,30 @@ def create(cls, tarball: Path, controller: "Controller") -> "Tarball":
 
         return cls(destination, controller)
 
+    def cache_map(self, tarball: Path):
+        """Builds Hierarchy structure of a Tarball in a Dictionary
+        Format.
+
+        Args:
+            tarball: path of an unpacked tarball
+        """
+        cmap = {tarball.name: {"details": TarInfo(tarball)}}
+        dir_queue = [(tarball, cmap)]
+        while dir_queue:
+            tar_path, parent_map = dir_queue.pop(0)
+            tar_n = tar_path.name
+            curr = parent_map[tar_n]
+
+            dir_list = tar_path.glob("*")
+            for l_path in dir_list:
+                tar_info = TarInfo(l_path)
+                if l_path.is_dir():
+                    curr[l_path.name] = {"details": tar_info}
+                    dir_queue.append((l_path, curr))
+                elif l_path.is_file:
+                    curr[l_path.name] = {"details": tar_info}
```

Comment:
directory, files and symlinks that is all that is included there right?

Reply:
I'd like to treat symlinks how they're treated on the filesystem. Essentially, the "path" points to the "real deal" whether it's the file/dir itself or the linked file/dir. That means always a single cached artifact, even if we have multiple cache map entries with the distinct aliased names.

Broken symlinks? Yeah; in effect, the Unix symlink is just a small text file, and can be (ab)used that way. So is preserving the name of the symlink and its "link" contents something we want to do? ☹️

Reply:
So, I guess there are two facets to this:  one is how we represent the cache map entries internally, and the other is how we present them to the caller.

I think it would be nice, at least initially, to "make symlinks disappear", until we have a requirement to do otherwise.  So, when we construct the map, when we hit a (valid) symlink, we can represent it as what it points to (with all the caveats above taken into account).  In this way, the representation and presentation are basically the same.

If we really need to represent (valid) symlinks separately from their referents, we can do that, but then we have to decide how to present them.  (And, again, I'm inclined to "make them disappear" in presentation.)

And, of course, we need to decide what to do with broken symlinks, but I'm OK (at least initially) with not representing them in the cache map (just as I presume we won't represent sockets, pipes, and other special files).